src/vm/jitinterface.cpp

```diff
@@ -13670,6 +13679,27 @@ BOOL LoadDynamicInfoEntry(Module *currentModule,
         }
         break;
 
+    case ENCODE_INDIRECT_PINVOKE_TARGET_SUPPRESS_GC:
+        {
+            MethodDesc *pMethod = ZapSig::DecodeMethod(currentModule, pInfoModule, pBlob);
+
+            _ASSERTE(pMethod->IsNDirect());
+            NDirectMethodDesc *pMD = (NDirectMethodDesc*)pMethod;
+
+            // Check if the NDirectTarget is marked for GC suppression at which point
+            // we must resolve the target at JIT time instead of at run time since there
+            // will be no GC frame and not in the correct mode.
+            void *tgtMaybe = nullptr;
+            if (NDirectMethodDesc::TryImportNDirectTargetForNoGCTransition(pMD, &tgtMaybe))
```

Comment:
This should not be calling TryImportNDirectTargetForNoGCTransition that does the custom attribute lookup. It should resolve the target unconditionally.

Reply:
I am being dense here. This code path is only for crossgen so the cost for lookup was done and properly computed prior. This was the missing point from the discussion above. I see what is being suggested and it makes sense. Thanks.

Reply:
Done.